components/playlist/browser/resources/reducers/playlist_reducer.ts

```diff
@@ -3,48 +3,51 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at https://mozilla.org/MPL/2.0/. */
 
-import { Playlist } from 'components/definitions/playlist'
+import * as States from 'components/playlist/browser/resources/reducers/states'
 import { Reducer } from 'redux'
 
 import { types } from '../constants/playlist_types'
 
 import * as PlaylistMojo from 'gen/brave/components/playlist/common/mojom/playlist.mojom.m.js'
 
-const playlistReducer: Reducer<Playlist.State | undefined> =
-    (state: Playlist.State | undefined, action) => {
-      if (state === undefined) {
-        state =
-          { lists: [], currentList: undefined, lastPlayerState: undefined }
+const playlistReducer: Reducer<States.PlaylistData | undefined> = (
+  state: States.PlaylistData | undefined,
+  action
+) => {
+  if (state === undefined) {
+    state = { lists: [], currentList: undefined, lastPlayerState: undefined }
+  }
+
+  switch (action.type) {
+    case types.PLAYLIST_LOADED:
+      const playlists = action.payload
+      let currentList: PlaylistMojo.Playlist | undefined
+      if (state.currentList) {
+        currentList = playlists.find((list: PlaylistMojo.Playlist) => {
+          return list.id === state?.currentList?.id
```

Comment:
At this point, state cannot be nullish (there's the state === undefined check).

```suggestion
          return list.id === state.currentList?.id
```

I'm a little confused about exactly what this is doing. Is there ever a case where the id in `currentList` is not the id for the playlist in currentList?

Reply:
Is this so we can dispatch a `PLAYLIST_SELECTED` event with just the id of the current list, and have it automatically load?